### PR TITLE
806 use Court Report Status instead of Court Report Submitted

### LINKED ISSUE #806: supervisors/admins can edit court report submission status
Part of epic #880 

**What type of user is this for? [volunteer/supervisor/admin/all or All CASA Admin]**
supervisors and admins

**Where does/should this occur?**
On the Edit CASA Case page (example: https://casa-qa.herokuapp.com/casa_cases/2/edit)
Screen Shot 2020-09-26 at 4 30 47 PM (2)

**Description**
As a supervisor/admin, I should be able to add the following statuses to `court_report_submission`:
- Submitted
- In review
- Completed

**Frontend**
There should be a "Select Status" dropdown menu next to Court Report Submission (see issue #619) where admins and supervisors can select from one of the statuses listed above. The status should then display under CASA Case details next to the text Court Report Submission, which displays for all user types viewing a `casa_case`'s details.

### PR DESCRIPTION
**Pull request by ChaelCodes**
### What github issue is this PR for, if any?
Resolves #806 

### What changed, and why?
`court_report_submitted` as a boolean only tracked whether or not the report had been submitted. It did not track when it was submitted nor if it was completed. This PR adds `court_report_status` which tracks the current status of the court report, and `court_report_submitted_at` which tracks when it was submitted.

Cases will now have a court report status of 'Not submitted' by default. All users can update `court_report_status`. When the `court_report_status` is set to 'Submitted', 'In review', or 'Completed' for the first time, the `court_report_submitted_at` field is immediately set. If the `court_report_status` is ever set to 'Not submitted', the `court_report_submitted_at` date field will be cleared.

This PR also adds an after_party backfill that modifies all cases that were previously marked as `court_report_submitted`. All cases will be updated to have a `court_report_submitted_at` of the current date and `court_report_status` of submitted.

### How will this affect user permissions?
- Volunteer permissions: No Change
- Supervisor permissions: No Change
- Admin permissions: No Change

### How is this tested? (please write tests!) 💖💪
- Using Unit tests, and updating the request specs for casa cases

### Screenshots
Details Page - Not Submitted (`court_report_status: :not_submitted, court_report_submitted_at: nil`)

Details Page - Submitted (`court_report_status: :submitted, court_report_submitted_at: '2020-11-08 18:29:16'`)

Details Page - Completed (`court_report_status: :completed, court_report_submitted_at: '2020-11-08 18:29:16'`)

Edit Page
image

### CODE REVIEW

File: lib/tasks/deployment/20201108181154_backfill_court_report_submitted_at.rake [hunk 1]
```diff
@@ -0,0 +1,14 @@
+namespace :after_party do
```
Comment (compwron) [inline lib/tasks/deployment/20201108181154_backfill_court_report_submitted_at.rake:1]: Yay backfill!

**Review by compwron — APPROVED** [review]
Very nice! :)

**Comment by ChaelCodes** [comment]
@compwron hey! This is awkward, I should've made this as a draft PR, but some of the specs still aren't passing in this version.

**Comment by compwron** [comment]
@ChaelCodes weird, the build looked green O.o maybe I am confused. Please make a new PR when you have a chance, I'll revert or xit on master.

**Comment by ChaelCodes** [comment]
@compwron I made a new PR - https://github.com/rubyforgood/casa/pull/1285 It should fix the build, but I'm still waiting on Travis.

**Pull request merged**